spring-auto-restdocs-dokka-json/src/test/kotlin/capital/scalable/dokka/json/JsonFormatIntegrationTest.kt

```diff
@@ -0,0 +1,117 @@
+import org.jetbrains.dokka.NodeKind
+import org.jetbrains.dokka.NodeLocationAwareGenerator
+import org.jetbrains.dokka.PackageListService
+import org.jetbrains.dokka.Platform
+import org.jetbrains.dokka.Utilities.DokkaRunModule
+import org.jetbrains.dokka.Utilities.ServiceLocator
+import org.junit.Test
+import java.io.File
+
+class JsonFormatIntegrationTest {
+
+    @Test
+    fun `Should load descriptor from properties file, bind and use it`() {
+        DokkaGenerator(DokkaConfigurationImpl(), DokkaConsoleLogger).generate()
+    }
+}
+
+data class SourceLinkDefinitionImpl(override val path: String,
+                                    override val url: String,
+                                    override val lineSuffix: String?) : DokkaConfiguration.SourceLinkDefinition {
+    companion object {
+        fun parseSourceLinkDefinition(srcLink: String): DokkaConfiguration.SourceLinkDefinition {
+            val (path, urlAndLine) = srcLink.split('=')
+            return SourceLinkDefinitionImpl(
+                File(path).canonicalPath,
+                urlAndLine.substringBefore("#"),
+                urlAndLine.substringAfter("#", "").let { if (it.isEmpty()) null else "#$it" })
+        }
+    }
+}
+
+class SourceRootImpl(path: String) : DokkaConfiguration.SourceRoot {
+    override val path: String = File(path).absolutePath
+
+    companion object {
+        fun parseSourceRoot(sourceRoot: String): DokkaConfiguration.SourceRoot = SourceRootImpl(sourceRoot)
+    }
+}
+
+data class PackageOptionsImpl(override val prefix: String,
+                              override val includeNonPublic: Boolean = false,
+                              override val reportUndocumented: Boolean = true,
+                              override val skipDeprecated: Boolean = false,
+                              override val suppress: Boolean = false) : DokkaConfiguration.PackageOptions
+
+class DokkaConfigurationImpl(
+    override val outputDir: String = "",
+    override val format: String = "auto-restdocs-json",
+    override val generateIndexPages: Boolean = false,
+    override val cacheRoot: String? = null,
+    override val impliedPlatforms: List<String> = emptyList(),
+    override val passesConfigurations: List<DokkaConfiguration.PassConfiguration> = listOf(PassConfigurationImpl())
+) : DokkaConfiguration
+
+class PassConfigurationImpl (
+    override val classpath: List<String> = emptyList(),
+    override val moduleName: String = "",
+    override val sourceRoots: List<DokkaConfiguration.SourceRoot> = emptyList(),
+    override val samples: List<String> = emptyList(),
+    override val includes: List<String> = emptyList(),
+    override val includeNonPublic: Boolean = false,
+    override val includeRootPackage: Boolean = false,
+    override val reportUndocumented: Boolean = false,
+    override val skipEmptyPackages: Boolean = false,
+    override val skipDeprecated: Boolean = false,
+    override val jdkVersion: Int = 6,
```

Comment:
6?

Reply:
It does not matter for the test but I changed it to 8 to have it in line with what we actually use.